src/julia-syntax.scm

```diff
@@ -1406,19 +1406,22 @@
             (reverse a))))))
 
 ;; lower function call containing keyword arguments
-(define (lower-kw-call f kw pa)
+(define (lower-kw-call fexpr kw pa)
   (let ((container (make-ssavalue)))
     (let loop ((kw kw)
                (initial-kw '()) ;; keyword args before any splats
                (stmts '())
                (has-kw #f))     ;; whether there are definitely >0 kwargs
       (if (null? kw)
-          (if (null? stmts)
-              `(call (call (core kwfunc) ,f) (call (top vector_any) ,@(reverse initial-kw)) ,f ,@pa)
-              `(block
-                (= ,container (call (top vector_any) ,@(reverse initial-kw)))
-                ,@(reverse stmts)
-                ,(if has-kw
+          (let ((f (make-ssavalue)))
+            `(block
+              (= ,f, fexpr)
```

Comment:
Would be good to insert this only if `(not (sym-ref? fexpr))`, to keep the IR simpler.